core/src/main/java/org/infinispan/container/entries/ReadCommittedEntry.java

```diff
@@ -115,6 +117,10 @@ public final Object getValue() {
       return value;
    }
 
+   public Object getOldValue() {
```

Comment:
Needs javadoc, is this the value at the start of the transaction/operation, or is it supposed to be updated after every operation on this entry in the tx or after every retry that invokes `setValue()`?

Reply:
I would say it is the current value stored in DataContainer when you access it for the first time.

Reply:
The ```oldValue``` was promoted from ```RepeatableReadEntry``` and it says 
```Value before the last modification. Serves as the previous value when the operation is retried```

If you agree with it, I will bring it to this method

Reply:
I'm fine with it, but is that what you need for the query interceptor? Don't you need the value that was in the data container when you first read the key, like Pedro said?

Reply:
> Don't you need the value that was in the data container when you first read the key,

Yes, that's pretty much it

Reply:
In ```RepeatableReadEntry``` oldValue behaves like the initialValue. There's also a method to update it: ```updatePreviousValue```